### PR TITLE
[dhctl] fix random sorting of master in ssh connections list

### PR DESCRIPTION
**Pull request by alexey-gavrilov-flant**
## Description

The new variant replaces the list of strings with a structure of name and ip that allows correct output of ip and master.

## Why do we need it, and what problem does it solve?
Previously the code relied on two lists and the fact that the items in them are sorted equally, which was incorrect because of the use of map when adding nodes to the list of ssh hosts. 

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: dhctl
type: fix
summary: fix random sorting of master in the list
impact_level: default
```

### CODE REVIEW

File: dhctl/pkg/app/ssh.go [hunk 1]
```diff
@@ -34,7 +36,8 @@ var (
 	SSHBastionPort       = ""
 	SSHBastionUser       = os.Getenv("USER")
 	SSHUser              = os.Getenv("USER")
-	SSHHosts             = make([]string, 0)
+	SSHHosts             = make([]session.Host, 0)
+	sSHHostsRaw          = make([]string, 0)
```
Comment (name212) [inline dhctl/pkg/app/ssh.go:40]:
```suggestion
	sshHostsRaw          = make([]string, 0)
```

File: dhctl/pkg/system/node/ssh/checks.go [hunk 2]
```diff
@@ -46,45 +58,43 @@ func CheckSSHHosts(userPassedHosts []string, nodesNames []string, runConfirm fun
 		warnMsg = notPassedWarn
 	case userPassedHostsLen < replicas:
 		warnMsg = notEnthoughtWarn
+	case userPassedHostsLen == 3 && replicas == 1 && phase == "scale-to-single-master":
```
Comment (name212) [inline dhctl/pkg/system/node/ssh/checks.go:61]: please add comment for this condition

**Review by name212 — CHANGES_REQUESTED** [review]
two small thing

**Review by name212 — APPROVED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Cherry pick PR [10691](https://github.com/deckhouse/deckhouse/pull/10691) to the branch [release-1.65](https://github.com/deckhouse/deckhouse/tree/release-1.65) successful!

**Comment by deckhouse-BOaTswain** [comment]
Cherry pick PR [10692](https://github.com/deckhouse/deckhouse/pull/10692) to the branch [release-1.66](https://github.com/deckhouse/deckhouse/tree/release-1.66) successful!

**Pull request merged**